flink-fs-tests/src/test/java/org/apache/flink/hdfstests/ContinuousFileProcessingMigrationTest.java

```diff
@@ -309,17 +288,34 @@ public void testMonitoringSourceRestore() throws Exception {
 
         testHarness.setup();
 
-        testHarness.initializeState(
-                OperatorSnapshotUtil.getResourceFilename(
-                        "monitoring-function-migration-test-"
-                                + expectedModTime
-                                + "-flink"
-                                + testMigrateVersion
-                                + "-snapshot"));
+        // Get the exact filename
+        Tuple2<String, Long> fileNameAndModTime = getResourceFilename(testMigrateVersion);
+
+        testHarness.initializeState(fileNameAndModTime.f0);
 
         testHarness.open();
 
-        Assert.assertEquals((long) expectedModTime, monitoringFunction.getGlobalModificationTime());
+        Assert.assertEquals(
+                fileNameAndModTime.f1.longValue(), monitoringFunction.getGlobalModificationTime());
+    }
+
+    private Tuple2<String, Long> getResourceFilename(FlinkVersion version) throws IOException {
+        String resourceDirectory = OperatorSnapshotUtil.getResourceFilename("");
+        Pattern fileNamePattern =
+                Pattern.compile(
+                        "monitoring-function-migration-test-(\\d+)-flink"
```

Comment:
I did some digging through the code. My impression is that we don't even need the modification time, do we? It only ends up in the filename :thinking:

Reply:
Since there is a check at the last of `testMonitoringSourceRestore`:

```java
Assert.assertEquals(
                fileNameAndModTime.f1.longValue(), monitoringFunction.getGlobalModificationTime());
```

Thus the modification time might still be required. 

But there might be another option that we do not store the modification time in the filename, instead we read it directly from the file's meta during tests. If it works then we could get rid of the mod time in the filename. 

Do you think it is ok for us to file a separate issue to further optimize this test?